### PR TITLE
Add module-write hook

### PR DESCRIPTION
**Pull request by Flamefire**
This hook will be called anytime a module is written, so even for the fake modules used for sanity checks etc.
It gets passed the easyblock as the first argument and the filename and content to be written as the next 2 and may return a new text to be used for the module file. If nothing or None is returned the original text will be used.

### CODE REVIEW

File: easybuild/tools/hooks.py [hunk 1]
```diff
@@ -162,18 +160,16 @@ def find_hook(label, hooks, pre_step_hook=False, post_step_hook=False):
...
     return res
 
 
 def run_hook(label, hooks, pre_step_hook=False, post_step_hook=False, args=None, msg=None):
     """
-    Run hook with specified label.
+    Run hook with specified label and return result of hook or None.
 
     :param label: name of hook
     :param hooks: list of defined hooks
```
Comment (smoors) [inline easybuild/tools/hooks.py:175]: here hooks is also a dict, right?

File: easybuild/tools/hooks.py [hunk 2]
```diff
@@ -120,10 +121,7 @@ def load_hooks(hooks_path):
 
 def verify_hooks(hooks):
     """Check whether list of obtained hooks only includes known hooks."""
```
Comment (smoors) [inline easybuild/tools/hooks.py:123]: here hooks is also a dict, right?
and also in easybuild/framework/easyblock.py line 147?

File: easybuild/tools/hooks.py [hunk 3]
```diff
@@ -162,18 +160,16 @@ def find_hook(label, hooks, pre_step_hook=False, post_step_hook=False):
...
+    if res:
+        _log.info("Found %s hook", hook_name)
 
     return res
 
 
 def run_hook(label, hooks, pre_step_hook=False, post_step_hook=False, args=None, msg=None):
     """
-    Run hook with specified label.
+    Run hook with specified label and return result of hook or None.
```
Comment (smoors) [inline easybuild/tools/hooks.py:172]:
```suggestion
    Run hook with specified label and return result of hook() or None.
```

**Review by smoors — COMMENTED** [review]
looks good to me, apart from some comment updates.

File: easybuild/tools/hooks.py (hunk 1 above)
Comment (Flamefire) [reply easybuild/tools/hooks.py]: Yes. Fixed

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @Flamefire!

**Pull request merged**